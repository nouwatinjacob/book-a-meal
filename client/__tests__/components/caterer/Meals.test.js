import React from 'react';
import {
  shallow
} from 'enzyme';
import sweetalert from 'sweetalert';
import {
  Meals
} from '../../../src/components/caterer/Meals.jsx';
import {
  mealResponse
} from '../../__mockData__/mockMeal';

const props = {
  deleteAction: jest.fn(),
  meals: mealResponse.meal
};

jest.mock('sweetalert');

describe('Meals component', () => {
  localStorage.setItem('token', catererToken);
  const setup = () => shallow(<Meals {...props} />);

  it('should render Meals component correctly', (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });

  it(`should call 'handleDelete'
  when modify order button is clicked`, (done) => {
    const wrapper = setup();

    const handleDeleteSpy = jest.spyOn(
      wrapper.instance(),
      'handleDelete'
    );
    sweetalert.mockResolvedValue(Promise.resolve(true));
    wrapper.instance().handleDelete();
    expect(handleDeleteSpy).toHaveBeenCalled();

    done();
  });
});