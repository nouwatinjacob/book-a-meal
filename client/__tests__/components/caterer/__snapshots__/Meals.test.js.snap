// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meals component should render Meals component correctly 1`] = `
<tr>
  <td>
    Fried Rice and chicken Lap
  </td>
  <td>
    <img
      height="50"
      src="https://res.cloudinary.com/sansaristic/image/upload/v1536197477/BookMeal/1536197474332pexels-photo-247685.png.png"
      width="60"
    />
  </td>
  <td>
    3000
  </td>
  <td>
    <button
      className="button warning"
      style={
        Object {
          "marginRight": "5px",
        }
      }
    >
      <Link
        replace={false}
        to="/edit-meal/1"
      >
        <i
          className="fa fa-pencil"
        />
      </Link>
    </button>
    <button
      className="button danger"
      onClick={[Function]}
    >
      <i
        className="fa fa-trash"
      />
    </button>
  </td>
</tr>
`;
